examples/text/CC100.ipynb

```diff
@@ -1,202 +1,211 @@
-                        ]
-                    },
-                    "metadata": {},
-                    "execution_count": 8,
-                }
-            ],
-            "metadata": {},
-        },
-    ],
-    "metadata": {
-        "kernelspec": {"display_name": "Python 3", "language": "python", "name": "python3"},
-        "language_info": {
-            "codemirror_mode": {"name": "ipython", "version": 3},
-            "file_extension": ".py",
-            "mimetype": "text/x-python",
-            "name": "python",
-            "nbconvert_exporter": "python",
-            "pygments_lexer": "ipython3",
-            "version": "3.9.5",
-        },
-    },
-    "nbformat": 4,
-    "nbformat_minor": 5,
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "source": [
+    "import torch\n",
+    "import os\n",
+    "\n",
+    "from torchdata.datapipes.iter import (\n",
+    "    FileOpener,\n",
+    "    HttpReader,\n",
+    "    IterableWrapper,\n",
+    "    SampleMultiplexer,\n",
+    ")\n",
+    "\n",
+    "ROOT_DIR = os.path.expanduser('~/.torchdata/CC100')  # This directory needs to be crated and set"
+   ],
+   "outputs": [],
+   "metadata": {}
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "source": [
+    "# CC100 support (http://data.statmt.org/cc-100/)\n",
+    "\n",
+    "URL=\"http://data.statmt.org/cc-100/%s.txt.xz\"\n",
+    "VALID_CODES = [\n",
+    "    \"am\", \"ar\", \"as\", \"az\", \"be\", \"bg\", \"bn\", \"bn_rom\", \"br\", \"bs\", \"ca\", \"cs\", \"cy\", \"da\", \"de\", \n",
+    "    \"el\", \"en\", \"eo\", \"es\", \"et\", \"eu\", \"fa\", \"ff\", \"fi\", \"fr\", \"fy\", \"ga\", \"gd\", \"gl\", \"gn\", \"gu\", \n",
+    "    \"ha\", \"he\", \"hi\", \"hi_rom\", \"hr\", \"ht\", \"hu\", \"hy\", \"id\", \"ig\", \"is\", \"it\", \"ja\", \"jv\", \"ka\", \n",
+    "    \"kk\", \"km\", \"kn\", \"ko\", \"ku\", \"ky\", \"la\", \"lg\", \"li\", \"ln\", \"lo\", \"lt\", \"lv\", \"mg\", \"mk\", \"ml\", \n",
+    "    \"mn\", \"mr\", \"ms\", \"my\", \"my_zaw\", \"ne\", \"nl\", \"no\", \"ns\", \"om\", \"or\", \"pa\", \"pl\", \"ps\", \"pt\", \n",
+    "    \"qu\", \"rm\", \"ro\", \"ru\", \"sa\", \"si\", \"sc\", \"sd\", \"sk\", \"sl\", \"so\", \"sq\", \"sr\", \"ss\", \"su\", \"sv\", \n",
+    "    \"sw\", \"ta\", \"ta_rom\", \"te\", \"te_rom\", \"th\", \"tl\", \"tn\", \"tr\", \"ug\", \"uk\", \"ur\", \"ur_rom\", \"uz\", \n",
+    "    \"vi\", \"wo\", \"xh\", \"yi\", \"yo\", \"zh-Hans\", \"zh-Hant\", \"zu\",    \n",
+    "]\n",
+    "\n",
+    "def CC100(root, language_code, use_caching=True):\n",
+    "    if language_code not in VALID_CODES:\n",
+    "        raise ValueError(f\"Invalid language code {language_code}\")\n",
+    "    url = URL % language_code\n",
+    "    if use_caching:\n",
+    "        cache_compressed_dp = HttpReader(cache_compressed_dp).map(lambda x: (x[0]))\n",
```

Comment:
@ejguan  Looks like we are missing initializing cache_compressed_dp :)

Reply:
I was trying to test the code before landing but unfortunately the server holding CC100 Dataset was unreachable for a long time. And, I plan to move this to a script rather than notebook in the short future.